src/python/dxpy/app_builder.py

```diff
@@ -539,6 +539,22 @@ def upload_applet(src_dir, uploaded_resources, check_name_collisions=True, overw
             applet_spec["runSpec"]["code"] = code_fh.read()
             del applet_spec["runSpec"]["file"]
 
+    # If this is applet requires a cluster, inline any bootstrapScript code that may be provided.
+    # bootstrapScript is an *optional* clusterSpec parameter.
+    # NOTE: assumes bootstrapScript is always provided as a filename
+    if "systemRequirements" in applet_spec["runSpec"]:
+        sys_reqs = applet_spec["runSpec"]["systemRequirements"]
+        for entry_point in sys_reqs:
+            try:
+                bootstrapScript = os.path.join(src_dir, sys_reqs[entry_point]["clusterSpec"]["bootstrapScript"])
```

Comment:
We can require the script names to follow the same format we use for `dx get`, with region prefix, etc. This way, it will be guaranteed the files generated by dx get have always the same names as the original files used for `build`.

Also, if the bootstrap script contents is not region-specific, why do we need a different script for each region? Wouldn't one file be enough? Maybe I'm missing something.

Reply:
Hmm, a thought after our discussion: requiring a specific name format during `dx build` doesn't make sense though since users will probably want to use one script stored locally for all or most regions.

Reply:
Yes, I agree, users should be able to write one script and reference it from multiple regions and/or entry points in the spec. `dx get` will write the script back into a file(s) that follows a standard naming scheme, but imposing a naming scheme when the user is writing the app spec isn't necessary.